### DISCUSSION TITLE
Unity UI Question

### DISCUSSION THREAD

**Discussion by Blackwolf7704** [SmartlyDressedGames/Unturned-3.x-Community#4330] [Q&A] [answered]
I created a boss zombie's health bar (at the top of the game) in Unity UI, but it doesn't work.

I know that in order for the health bar to naturally shrink, the fill value of the image needs to be reduced, but looking at related posts, it doesn't seem like this type of UI works.

And I learned from the previous Q&A that when creating a custom script in Unity, you have to use a function that exists inside the game.
Am I understanding this correctly?

For example below.

Code in https://github.com/Unturned-Datamining/Unturned-Datamining
```
using System;
using UnityEngine;

namespace SDG.Unturned;

public class Acid: MonoBehaviour
{
      private bool isExploded;

      private Vector3 lastPos;

      publicGuid effectGuid;
.....
}
```

acid.cs code from olie's repository
```
using UnityEngine;

namespace SDG.Unturned
{
      // Plays the specified effect when something collides with the Collider component on the base Game Object
      // Destroys itself after one use and cannot be reset
      public class Acid: MonoBehaviour
      {
          public ushort effectID;
      }
}
```

**Comment by DanielWillett** [comment]
Yes, you can't add any code to the game through mods unless you do it purely through unity's UI (using mod hooks and events). The way most people do bars like that is by using the unicode block character in a text component.

Another couple of options:
1. Custom font (better performance compared to the second one)
2. A set number of images for different states of the bar
(example one image for every 2%, then enable, say, the first 20 of them when the bar's at 40%)

**Reply by Blackwolf7704** [reply]
I never thought about custom fonts. It helped me a lot!

Mode hooks and events are not defined by Nelson, but can only some of them be imported and used like acid.cs above?

**Reply by DanielWillett** [reply, marked as answer]
ModHooks are a select number of classes Nelson exports for use, but you can technically export any and use them, just most won't be useful or won't work properly.

**Reply by Blackwolf7704** [reply]
Then, it seems that the mod hook provided by Nelson is the only thing that can be implemented reliably in Unity. thank you
